### ISSUE TITLE
`no-duplicate-landmark-elements` should not flag elements as duplicate across different conditional branches

### ISSUE THREAD

**Issue by bmish** [ember-template-lint/ember-template-lint#1799]
`<header>` is flagged as a violation of [no-duplicate-landmark-elements](https://github.com/ember-template-lint/ember-template-lint/blob/master/docs/rule/no-duplicate-landmark-elements.md) in this example:

```hbs
{{#if @someVariable}}
  <header>
    foo
  </header>
{{else}}
  <header>
    bar
  </header>
{{/if}}
```

I know that `<header>` could be moved outside the if statement in this simplified example, but that fix is not always possible. We should ignore duplicates across conditional branches.

The `no-duplicate-id` rule already takes this same problem into consideration.

ember-template-lint: `3.0.0-beta.5`

**Comment by rwjblue** [comment]
Hmm, I thought we landed a fix for this recently. 🤔

**Comment by MelSumner** [comment]
> Hmm, I thought we landed a fix for this recently. 🤔

me too.

**Comment by rwjblue** [comment]
We fixed no-duplicate-id, it's the same issue and likely the same fix will work.

**Comment by JimSchofield** [comment]
I have a code snippet where the second and third `main` elements are being flagged on different conditional branches:
```html
{{#if this.shouldGenerateToc}}
  <div class="flex gap-5">
    ...
    <main class="w-4/5">
      {{yield}}
    </main>
  </div>
{{else if this.shouldUsePortal}}
  <div class="flex gap-5">
    ...
    <main class="w-4/5">
      {{yield}}
    </main>
  </div>
{{else}}
  <div class="flex gap-5">
    <main class="w-full">
      {{yield}}
    </main>
  </div>
{{/if}}
```

**Comment by gossi** [comment]
Run into this as well, here is my relevant snippet:

```hbs
{{#let (unique-id) as |runtimeId|}}
<section local-class="section" aria-labelledby={{runtimeId}}>
	<header id={{runtimeId}}>
    	{{t "a.b.c"}}
	</header>

	...
</section>
{{/let}}

{{#let (unique-id) as |paymentId|}}
<section local-class="section" aria-labelledby={{paymentId}}>
	<header id={{paymentId}}>
    	{{t "a.b.d"}}
	</header>

	...
</section>
{{/let}}
```

**Comment by lifeart** [comment]
fix should be close to https://github.com/ember-template-lint/ember-template-lint/pull/3176
